great_expectations/expectations/metrics/table_metrics/table_column_types.py

```diff
@@ -47,21 +47,12 @@ def _sqlalchemy(
         metrics: Dict[Tuple, Any],
         runtime_configuration: Dict,
     ):
-        batch_id = metric_domain_kwargs.get("batch_id")
-        if batch_id is None:
-            if execution_engine.active_batch_data_id is not None:
-                batch_id = execution_engine.active_batch_data_id
-            else:
-                raise GreatExpectationsError(
-                    "batch_id could not be determined from domain kwargs and no active_batch_data is loaded into the "
-                    "execution engine"
-                )
-        batch_data = execution_engine.loaded_batch_data_dict.get(batch_id)
-        if batch_data is None:
-            raise GreatExpectationsError(
-                "the requested batch is not available; please load the batch into the execution engine."
-            )
-        return _get_sqlalchemy_column_metadata(execution_engine.engine, batch_data)
+        batch_data, _, _ = execution_engine.get_compute_domain(
```

Comment:
👍 The above error checking is already contained in `get_compute_domain()`

Reply:
@anthonyburdi Yes -- and even more than error checking.  This is moving the metric implementation to use the correct interface -- the very subject of this PR.  Thanks!